src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -184,6 +205,28 @@ public void Reanalyze(IIncrementalAnalyzer analyzer, ReanalyzeScope scope, bool
                 }
             }
 
+            private void OnActiveDocumentChanged(object sender, DocumentId activeDocumentId)
+            {
+                IAsyncToken asyncToken;
+                if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(_registration.Workspace.Options) == BackgroundAnalysisScope.ActiveFile &&
+                    activeDocumentId != null)
+                {
+                    var activeDocument = _registration.Workspace.CurrentSolution.GetDocument(activeDocumentId);
+                    if (activeDocument != null)
+                    {
+                        var lastActiveDocument = Interlocked.Exchange(ref _lastActiveDocument, activeDocument);
```

Comment:
all my comments above about just locking still apply. i almost never understand the desire to use hte interlocked stuff and then ahve to do all the mental gymnastics to ensure correctness.